### PR TITLE
Scan Reports Layout improvement

### LINKED ISSUE #800: Scan Report Layout Improvements
We need to utilise Nextjs layouts, and consistency across the design.

So one layout for a scan report, and the various pages beneath this are just elements on a scan report menu.

Data sets will be in a future PR.

Provides consistency in navigation, and an optimised experience.

- [x] Single reusable Breadcrumbs component
- [x] Layout for Scan Reports

### PR DESCRIPTION
**Pull request by AndrewThien**
# Changes

This PR adds some improvements to the layout of Scan-Reports-related pages, making it more accessible and cleaner. It also:
- Moved the Summary Rules View to a separate location (Review Rules)
- Adjusted some margin and padding in different views
- Cleaned repeated Breadcrumbs components 
- Re-organised Mapping Rules table and fix one linking bug there
- Update dependencies for the Next client

Closes #800 

# Screenshots

Screenshot 2024-07-26 at 14 24 56

Screenshot 2024-07-26 at 14 48 21

# Checks

**Important:** please complete these **before** merging.
- [x] Run migrations, if any.
- [x] Update `changelog.md`, including migration instructions if any.
- [x] Run unit tests.

### CODE REVIEW

File: app/next-client-app/app/datasets/page.tsx [hunk 1]
```diff
@@ -19,6 +19,7 @@ interface DataSetListProps {
 }
 
 export default async function DataSets({ searchParams }: DataSetListProps) {
+  // TODO: ADD default page size here
```
Comment (AndyRae) [inline app/next-client-app/app/datasets/page.tsx:22]: How come?

**Review by AndyRae — APPROVED** [review]

File: app/next-client-app/app/datasets/page.tsx (hunk 1 above)
Comment (AndrewThien) [reply app/next-client-app/app/datasets/page.tsx:22]: Yeah, sorry. The default page size, already defaulted by 10,  is not needed here. I will remove this TODO in the next PR. Thanks

File: app/next-client-app/app/scanreports/[id]/layout.tsx [hunk 2]
```diff
@@ -24,7 +63,130 @@ export default async function ScanReportLayout({
...
+          <div className="flex items-center">
+            <h3 className="text-gray-500">
+              Dataset:{" "}
+              <span className="text-black">{scanreport.parent_dataset}</span>
+            </h3>
+          </div>
+          <div>|</div>
+          <div className="flex items-center">
+            <h3 className="text-gray-500">
+              Data Partner:{" "}
```
Comment (AndyRae) [inline app/next-client-app/app/scanreports/[id]/layout.tsx:114]: For the future, we will make these link to their id pages.

**Review by AndyRae — CHANGES_REQUESTED** [review]

File: app/next-client-app/components/ui/layout/tab.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+  const Icon = item.iconName ? iconMap[item.iconName] : null;
+
+  return (
+    <>
+      {" "}
+      <Link href={href}>
+        <Button
+          variant={"ghost"}
+          className={cn("rounded-md px-0 py-1 text-xl", {
```
[the remark this answers is not in the document]
Comment (AndrewThien) [reply app/next-client-app/components/ui/layout/tab.tsx]: I agree that "text-xl" is a bit much, but considering that these are buttons without borders and padding like other button variants, I feel like the default text size will be a bit small and unclear for these nav-buttons. So, I am currently putting it as "text-lg". Please see below for references. If we still feel default size is still good, I will change it again 

- default size
Screenshot 2024-07-29 at 09 57 35

- large size
Screenshot 2024-07-29 at 09 56 22

File: app/next-client-app/components/ui/layout/scanreport-status.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  id,
+  status,
+  dataset,
+}: {
+  id: number;
+  status: string;
+  dataset: string;
+}) {
+  return (
+    <ScanReportStatus id={id} status={status} dataset={dataset} onLayout />
```
[the remark this answers is not in the document]
Comment (AndrewThien) [reply app/next-client-app/components/ui/layout/scanreport-status.tsx:15]: The original motive to have this separate "Client" component is to solve the fact that <ScanReportStatus> Component can't actually run directly on a page/component without "use Client" (see the error below). So I had to make a "Client" component as a parent for <ScanReportStatus> and then use this in the layout.tsx

[the remark this answers is not in the document]
Comment (AndrewThien) [reply app/next-client-app/components/ui/layout/scanreport-status.tsx:15]: Screenshot 2024-07-29 at 12 23 10

File: app/next-client-app/components/ui/layout/tab.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (AndyRae) [reply app/next-client-app/components/ui/layout/tab.tsx]: I think we should stick to the default size - using open metadata for our inspiration again.

File: app/next-client-app/components/ui/layout/scanreport-status.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (AndrewThien) [reply app/next-client-app/components/ui/layout/scanreport-status.tsx]: That makes sense. Pushed the changes. Thanks

File: app/next-client-app/components/ui/layout/tab.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (AndyRae) [reply app/next-client-app/components/ui/layout/tab.tsx]: Yes, that is perfect - thank you.

**Review by AndyRae — APPROVED** [review]

**Review by Dev-Akashili — APPROVED** [review]

**Pull request merged**